### PR TITLE
Boolean documentation updates

### LINKED ISSUE #231: Github requires boolean values for true/false values not 0/1
https://github.com/plu/Pithub/blob/cd85ed31181c5dc6df14f97852de894c1d2a790b/lib/Pithub/Repos/Releases.pm#L113

I had errors attempting to create a release when using 0 or 1 for the true false values as documented.

I have to do the following:

```
use constant false => \0;
sub true () { return \1 } # note the empty prototype

```
Then I had to use 'true' or 'false' for the attributes that required boolean values.

### PR DESCRIPTION
**Pull request by timlegge**
Fixes #231 
Replaces #232 

I left in the tests because they should still work out if someone runs the tests with the token enable,

### CODE REVIEW

**Review by oalders — APPROVED** [review]
🙏

**Comment by timlegge** [comment]
I pushed a fix for  "my" variable $result masks earlier declaration 
Not sure about the bare word issue

File: t/live/repos.t [hunk 1]
```diff
@@ -467,6 +467,39 @@ SKIP: {
...
+        # Using JSON true/false values
+        require JSON::MaybeXS;
+        my $result = $p->repos->releases->create(
+            data => {
+                tag_name         => "v0.01000",
+                target_commitish => "master",
+                name             => "This is a test release of v0.01000",
+                body             => "Here be notes",
+                draft            => JSON::MaybeXS::false,
+                prerelease       => JSON::MaybeXS::false,
```
Comment (oalders) [inline t/live/repos.t:481]: Changing these to `JSON::MaybeXS::false()` may fix the error.

Comment (timlegge) [reply t/live/repos.t]: @oalders done lets try again.

File: lib/Pithub/Repos/Releases.pm [hunk 2]
```diff
@@ -110,11 +110,57 @@ Examples:
...
+=item JSON::MaybeXS
+
+Add the following to your code before the call to Pithub::Repos::Releases->create.
+
+    require JSON::MaybeXS;
+
+Then use the following values for the booleans:
+
+    JSON::MaybeXS::true
+    JSON::MaybeXS::false
```
Comment (oalders) [inline lib/Pithub/Repos/Releases.pm:138]: I wonder if we should change these to the function calls, since we know those will work more consistently?

Comment (timlegge) [reply lib/Pithub/Repos/Releases.pm:138]: good idea - give me a few minutes

Comment (timlegge) [reply lib/Pithub/Repos/Releases.pm]: @oalders I think I found all of them

**Review by oalders — APPROVED** [review]
Thanks for all of the iterations @timlegge!

**Pull request merged**